airflow/providers/google/cloud/hooks/dataform.py

```diff
@@ -263,9 +301,26 @@ def cancel_workflow_invocation(
             f"projects/{project_id}/locations/{region}/repositories/"
             f"{repository_id}/workflowInvocations/{workflow_invocation_id}"
         )
-        client.cancel_workflow_invocation(
-            request={"name": name}, retry=retry, timeout=timeout, metadata=metadata
-        )
+        try:
+            workflow_invocation = self.get_workflow_invocation(
+                project_id=project_id,
+                region=region,
+                repository_id=repository_id,
+                workflow_invocation_id=workflow_invocation_id,
+            )
+            state = workflow_invocation.state
+        except Exception as err:
+            raise AirflowException(f"Dataform API returned error when waiting for workflow invocation:\n{err}")
+
+        if state == WorkflowInvocation.State.RUNNING:
+            client.cancel_workflow_invocation(
+                request={"name": name}, retry=retry, timeout=timeout, metadata=metadata
+            )
+        else:
+            self.log.info(
+                "Workflow is not active. Either the execution has already finished or has been canceled. "
+                "Please check the logs above for more details."
+            )
```

Comment:
Could you also cover these cases in unit tests please ?
1. get_workflow_invocation() throws an exception
2. state == WorkflowInvocation.State.RUNNING
3. state != WorkflowInvocation.State.RUNNING

Reply:
Done